### ISSUE TITLE
Generating reports with RF7+ new style output.xml

### ISSUE THREAD

**Issue by ArekKuczynski** [damies13/rfswarm#303]
**Describe the bug**
Reporter sometimes has problems generating html/docx/xlsx report files. This mainly affects large reports.

**To Reproduce**
1. Open Reporter
2. Load some results
3. Generate html/docx/xlsx

**Expected behavior**
Reporter should generate any reports without throwing exceptions.
We can also block the user from generating reports before the Preview is loaded.

Example exception:
```
Robot Framework Swarm: Reporter 
        Version 1.4.0
        Configuration File:  X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\RFSwarmReporter.ini
Exception in thread Thread-35 (cg_export_xhtml):
Traceback (most recent call last):
  File "C:\Users\arekk\AppData\Local\Programs\Python\Python39\Lib\threading.py", line 1073, in _bootstrap_inner
    self.run()
  File "C:\Users\arekk\AppData\Local\Programs\Python\Python39\Lib\threading.py", line 1010, in run
    self._target(*self._args, **self._kwargs)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 9910, in cg_export_xhtml
    core.export_xhtml()
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3270, in export_xhtml
    self.xhtml_add_sections(body, "TOP", sectionpct)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3488, in xhtml_add_sections
    self.xhtml_add_sections(nextparent, sect, sectionpct)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3481, in xhtml_add_sections
    self.xhtml_sections_errors(thiselmt, id)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3894, in xhtml_sections_errors
    base.rt_errors_get_data(id)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 2399, in rt_errors_get_data
    base.rt_errors_parse_xml(id, rid)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 2421, in rt_errors_parse_xml
    lfol = glob.glob(gpatt, recursive=True)[0]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

**Screenshots**

**Desktop (please complete the following information):**
- OS: Windows
- Reporter Version: 1.4.0 newest 30/09/2024

**Additional context**

**Comment by damies13** [comment]
Part of the fix for glob issues:

```
	def rt_errors_parse_xml(self, id, rid):
....
			iter = rdata['iteration']
			base.debugmsg(8, "iter:", iter)
			if '_' in iter:
				iter = iter.split('_')[0]

			base.debugmsg(8, "iter:", iter)
```
Then
```
			gpatt = os.path.join(ldir, "{}_{}_{}_*_{}*".format(rdata['script'].split('.')[0], rdata['script_index'], rdata['robot'], rdata['iteration']))
```
becomes
```
			gpatt = os.path.join(ldir, "{}_{}_{}_*_{}_1*".format(rdata['script'].split('.')[0], rdata['script_index'], rdata['robot'], iter))
```

And also
```
			gxpatt = os.path.join(ldir, "{}_{}_{}_*_{}*".format(rdata['script'].split('.')[0], rdata['script_index'], rdata['robot'], rdata['iteration']), "*_output.xml")
```
becomes
```
			gxpatt = os.path.join(ldir, "{}_{}_{}_*_{}_1*".format(rdata['script'].split('.')[0], rdata['script_index'], rdata['robot'], iter), "*_output.xml")
```


Then the correct xml files will get read

**Comment by damies13** [comment]
Fixing this was necessary to finish the development of issue  #139, so the fix for this (above) has already been merged into branch 1.4.0.

I believe this also is why we can no longer reproduce issue #304
